test/manifest.ts

```diff
@@ -0,0 +1,183 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {describe, it, afterEach} from 'mocha';
+
+import {Manifest} from '../src/manifest';
+import * as sinon from 'sinon';
+import {GitHub} from '../src/github';
+import {stubFilesFromFixtures} from './releasers/utils';
+import {expect} from 'chai';
+import {CheckpointType} from '../src/util/checkpoint';
+
+const fixturePath = './test/fixtures/manifest/repo';
+const defaultBranch = 'main';
+const sandbox = sinon.createSandbox();
+
+describe('Manifest', () => {
```

Comment:
For another TypeScript project I've been working on, my colleague @SurferJeffAtGoogle defined a schema using JSONSchema for our config format:

https://github.com/googleapis/repo-automation-bots/blob/master/packages/owl-bot/src/owl-bot-yaml-schema.json

and then used [ajv](https://www.npmjs.com/package/ajv) to perform validation. I thought this was a neat approach we might consider, since our manifest/config format is relatively complex.